plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/TypeEmbedding.kt

```diff
@@ -287,11 +288,11 @@ data class ClassTypeEmbedding(val className: ScopedKotlinName) : TypeEmbedding {
     fun getterFunctions(): List<FieldAccessFunction> {
         val receiver = PlaceholderVariableEmbedding(GetterFunctionSubjectName, this)
         val getPropertyFunctions = fields.values
-            .filter { field -> field.accessPolicy != AccessPolicy.ALWAYS_INHALE_EXHALE }
+            .filter { field -> field.accessPolicy == AccessPolicy.ALWAYS_READABLE }
             .map { field -> FieldAccessFunction(name, field, FieldAccess(receiver, field).pureToViper()) }
         val getSuperPropertyFunctions = classSuperTypes.flatMap {
             it.flatMapUniqueFields { _, field ->
-                if (field.accessPolicy != AccessPolicy.ALWAYS_INHALE_EXHALE) {
+                if (field.accessPolicy == AccessPolicy.ALWAYS_READABLE) {
```

Comment:
Is this code broken as-is, or is this necessary because of some higher-level change?

Reply:
It's necessary because special mutable fields are no longer included in the predicate that now is completely immutable. Anyway, I think these pure getter functions will be removed soon and then replaced by nested unfolding expressions